### ISSUE TITLE
Handle shared billing account effect on no IAM inheritance into billing roles for service accounts - IAM will be set, billing will not

### ISSUE THREAD

**Issue by obriensystems** [GoogleCloudPlatform/pbmm-on-gcp-onboarding#177]
TL;DR; Shared billing accounts do not get shared IAM roles - they need to be set separately

see (needs update for shared billing)
https://cloud.google.com/billing/docs/how-to/billing-access

Keep in mind that we need a workaround  for the fact that if the billing account is of type "shared" where it comes in under "No Organization, ID=0" then any service account created will not get inherited links from IAM set in Billing - these like Billing Account User - need to be set manually.

Example
```
michael@cloudshell:~$ gcloud config set project gcp-zone-landing-stg
Updated property [core/project].
michael@cloudshell:~ (gcp-zone-landing-stg)$ export PROJECT_ID=$(gcloud config list --format 'value(core.project)')
michael@cloudshell:~ (gcp-zone-landing-stg)$ export ORG_ID=$(gcloud projects get-ancestors $PROJECT_ID --format='get(id)' | tail -1)
michael@cloudshell:~ (gcp-zone-landing-stg)$ export SA_PREFIX=tfsa-example
michael@cloudshell:~ (gcp-zone-landing-stg)$ gcloud iam service-accounts create "${SA_PREFIX}" --display-name "Terraform example service account" --project=${PROJECT_ID}
Created service account [tfsa-example].
michael@cloudshell:~ (gcp-zone-landing-stg)$ export SA_EMAIL=`gcloud iam service-accounts list --project="${PROJECT_ID}" --filter=tfsa --format="value(email)"`
michael@cloudshell:~ (gcp-zone-landing-stg)$ echo $SA_EMAIL
tfsa-example@gcp-zone-landing-stg.iam.gserviceaccount.com

check existing roles
michael@cloudshell:~ (gcp-zone-landing-stg)$ gcloud organizations get-iam-policy $ORG_ID --filter="bindings.members:$SA_EMAIL" --flatten="bindings[].members" --format="table(bindings.role)"

Set the billing role
gcloud organizations add-iam-policy-binding ${ORG_ID}  --member=serviceAccount:${SA_EMAIL} --role=roles/billing.user

check again

michael@cloudshell:~ (gcp-zone-landing-stg)$ gcloud organizations add-iam-policy-binding ${ORG_ID}  --member=serviceAccount:${SA_EMAIL} --role=roles/billing.user
Updated IAM policy for organization [925207728429].
...
michael@cloudshell:~ (gcp-zone-landing-stg)$ gcloud organizations get-iam-policy $ORG_ID --filter="bindings.members:$SA_EMAIL" --flatten="bindings[].members" --format="table(bindings.role)"

ROLE: roles/billing.user
```
It may take a couple min to show in IAM

<img width="1434" alt="Screen Shot 2022-09-18 at 19 01 36" src="https://user-images.githubusercontent.com/24765473/190931947-eed320aa-77af-440d-bfa2-0c944c6a3fa9.png">

Checking billing on the shared account

expected on billing accounts belonging to this org - via IAM inheritance in billing
<img width="1737" alt="Screen Shot 2022-09-18 at 19 04 02" src="https://user-images.githubusercontent.com/24765473/190931997-e6cc1729-1046-4a49-835a-bfa86d25c09b.png">

not expected on billing accounts shared from other orgs
<img width="1738" alt="Screen Shot 2022-09-18 at 19 04 49" src="https://user-images.githubusercontent.com/24765473/190932022-f0e48504-9e95-4bba-aff6-8cf18e76eb26.png">

**Comment by fmichaelobrien** [comment]
20240406: Closing issue during retrofit/rebase of this TEF V1 based/modified repo to TEF V4 standards
This issue may participate in the LZ refactor after rebase
Query on all issues related to the older V1 version via the tag
https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/labels/2024-pre-tef-v4
